core/capabilities/ccip/configs/solana/chain_writer.go

```diff
@@ -261,6 +297,21 @@ func GetSolanaChainWriterConfig(offrampProgramAddress string, fromAddress string
 	return solConfig, nil
 }
 
+func getExecuteMethodConfigByVersion(version *string) ExecuteMethodConfigFunc {
+	if version == nil {
+		return getExecuteMethodConfigV1
+	}
+	versionStr := *version
+	switch versionStr {
+	case types.SolanaChainWriterExecuteConfigVersionV1:
+		return getExecuteMethodConfigV1
+	case types.SolanaChainWriterExecuteConfigVersionV2:
+		return getExecuteMethodConfigV2
+	default:
+		return getExecuteMethodConfigV1
+	}
+}
```

Comment:
Nice :+1:

Reply:
nit: you could have used a string here and skipped the ptr dereference and nil check

Reply:
I was considering it but since we default to something on nil I thought it made sense to encapsulate all of that within the method

Reply:
I guess thats true, but even if you were to set the value to "v3" it would still default to v1.

Reply:
Oh ya good point. I probably should have maybe explicitly handle "" for v1 and error in the default case for unsupported version